src/SFA.DAS.EmployerIncentives.Api/Extensions/HttpResponseExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using SFA.DAS.EmployerIncentives.Infrastructure.Api;
+
+namespace SFA.DAS.EmployerIncentives.Api.Extensions
+{
+    public static class HttpResponseExtensions
+    {
+        public static Task WriteJsonAsync(this HttpResponse httpResponse, object body)
+        {
+            httpResponse.ContentType = "application/json";
+
+            return httpResponse.WriteAsync(JsonConvert.SerializeObject(body, new JsonSerializerSettings
```

Comment:
Avoid using Newtonsoft.Json to load whole document into memory.  Use System.Text.Json that is part of core 3.1 instead to stream the data.

Reply:
agreed. Will remove